snapshotter/demux/metrics/proxy.go

```diff
@@ -30,10 +32,71 @@ const (
 	MaxMetricsResponseSize = 32768 // 32 KB
 )
 
+// Monitor is used for port and lifecycle management of metrics proxies.
+type Monitor struct {
+	// monitorChan indicates a port is free for metrics proxy server.
+	ch chan int
+	// portMap tracks used and unused metrics proxy server ports.
+	portMap map[int]bool
+	// startPort is first port in the configurable range of ports to serve metrics proxies from.
+	startPort int
+	// endPort is last port in the configurable range of ports to serve metrics proxies from.
+	endPort int
+	// mu is the lock used for threadsafe reading and writing of the metrics proxy port map.
+	mu *sync.Mutex
+}
+
+// NewMonitor returns a new metrics proxy monitor.
+func NewMonitor(startPort, endPort int) *Monitor {
+	return &Monitor{make(chan int), make(map[int]bool), startPort, endPort, &sync.Mutex{}}
+}
```

Comment:
Might be best to add some port range validation. start < end and start > 0

Reply:
I'll add, thanks.

Reply:
Yes, but maybe better to fail fast choosing to fail at monitor start vs failing on proxy object initialization. Thoughts?